core/src/main/scala/com/rallyhealth/vapors/core/time/ModifyTime.scala

```diff
@@ -0,0 +1,52 @@
+package com.rallyhealth.vapors.core.time
+
+import java.time.{Duration, Instant, LocalDate, LocalDateTime, Period, ZonedDateTime}
+import java.time.temporal.{Temporal, TemporalAmount}
+
+trait ModifyTime[T, -D] {
+
+  def addTo(
+    temporal: T,
+    amount: D,
+  ): T
+
+  def subtractFrom(
+    temporal: T,
+    amount: D,
+  ): T
+}
+
+object ModifyTime {
+
+  def addTo[T, D](
+    temporal: T,
+    amount: D,
+  )(implicit
+    modifier: ModifyTime[T, D],
+  ): T = modifier.addTo(temporal, amount)
+
+  def subtractFrom[T, D](
+    temporal: T,
+    amount: D,
+  )(implicit
+    modifier: ModifyTime[T, D],
+  ): T = modifier.subtractFrom(temporal, amount)
+
+  private final class OfTemporal[T <: Temporal, -D <: TemporalAmount] extends ModifyTime[T, D] {
+    override def addTo(
+      temporal: T,
+      amount: D,
+    ): T = amount.addTo(temporal).asInstanceOf[T]
+    override def subtractFrom(
+      temporal: T,
+      amount: D,
+    ): T = amount.subtractFrom(temporal).asInstanceOf[T]
```

Comment:
This isn't used. I should probably remove it.

Reply:
Agreed.